truss/templates/server/truss_server.py

```diff
@@ -79,7 +79,12 @@ def check_healthy(model: ModelWrapper):
             raise errors.ModelNotReady(model.name)
 
     async def model_ready(self, model_name: str) -> Dict[str, Union[str, bool]]:
-        self.check_healthy(self._safe_lookup_model(model_name))
+        model: ModelWrapper = self._safe_lookup_model(model_name)
+        is_ready = await model.is_ready()
+        if is_ready is None:
+            self.check_healthy(model)
```

Comment:
`check_healthy` is also called in `predict` here https://github.com/basetenlabs/truss/pull/1283/files#diff-b351a6c89f1ba008f68df0a7986f0f582a5fd9c96d61fb1dd9d0b553496d803cR160 to ensure the model has finished loading before running predict, I think we can leave this as is

Reply:
Did we change the implementation of `check_healthy` ?

Reply:
Nope, `check_healthy` remains the same and will continue to be called when a `is_healthy` health check isn't defined by a truss user

Reply:
Should we call `check_healthy` first? i.e. before calling the `model.is_healthy()`

Reply:
We return early from `is_healthy` if load fails, after which we will run `check_healthy` to ensure we still fail fast on load failure, this is the only scenario where we need to run `check_healthy` in addition to `is_healthy`, otherwise we should only call `is_healthy` since the custom health check should no longer just check if the model has loaded successfully

Reply:
cool, makes sense. Thanks